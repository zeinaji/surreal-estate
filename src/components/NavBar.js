import React from "react";
import "../styles/NavBar.css";
import logo from "../property.png";
import { Link } from "react-router-dom";

function NavBar() {
  return (
    <>
      <div className="wrapper">
        <nav className="navbar">
          <img className="logo" src={logo} alt="logo" />
          <ul className="navbar-links">
            <li className="navabar-links-item surreal-estate">
              Surreal Estate
            </li>
            <Link className="navabar-links-item" to="/">
              View Properties
            </Link>
            <Link className="navabar-links-item" to="/add-property">
              Add a Property
            </Link>
          </ul>
        </nav>
      </div>
    </>
  );
}

export default NavBar;
