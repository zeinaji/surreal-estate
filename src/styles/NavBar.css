.navbar {
  background: white;
  width: 100%;
  height: 70px;
  position: relative;
  display: flex;
  align-items: center;
}
.wrapper {
  width: 100%;
  height: 70px;
  padding: 0;
  padding-bottom: 3px;
  background: linear-gradient(to right, rgb(112, 201, 253), rgb(235, 149, 206));
}

.logo {
  margin: 10px 0px 10px 10px;
  width: 40px;
  height: 40px;
}

.navbar-links {
  display: flex;
  margin-top: 33px;
  list-style: none;
  margin-left: 0;
  padding-left: 0;
}

a {
  font-weight: 800;
  color: rgb(83, 80, 80);
  margin-right: 20px;
  text-decoration: none;
}

.surreal-estate {
  font-weight: 800;
  color: rgb(83, 80, 80);
}

a:hover {
  background: -webkit-linear-gradient(rgb(112, 201, 253), rgb(235, 149, 206));
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.surreal-estate {
  padding-left: 5px;
  padding-right: 50px;
}
