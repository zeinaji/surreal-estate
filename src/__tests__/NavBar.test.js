import React from "react";
import NavBar from "../components/NavBar";
import { render } from "@testing-library/react";

describe("NavBar", () => {
  xit("renders correctly", () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
