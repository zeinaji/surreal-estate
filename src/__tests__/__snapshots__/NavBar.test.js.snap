// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="navbar"
    >
      <img
        alt="logo"
        class="logo"
        src="property.png"
      />
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item surreal-estate"
        >
          Surreal Estate
        </li>
        <li
          class="navbar-links-item"
        >
          View Properties
        </li>
        <li
          class="navbar-links-item"
        >
          Add a Property
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
